src/saltext/vmware/states/vmc_sddc.py

```diff
@@ -0,0 +1,397 @@
+"""
+VMC-SDDC state module
+
+Add new SDDC and delete existing SDDC.
+
+Example usage :
+
+.. code-block:: yaml
+
+    ensure_sddc_exists:
+        vmc_sddc.present:
+            - hostname: stg.skyscraper.vmware.com
+            - refresh_key: bRfsZYg3vVFOZEBHDp9GFSKv76UMfc4DOnWgNk9vDxNVPfHqt73rk75x0vSUtx8w
+            - authorization_host: console-stg.cloud.vmware.com
+            - org_id: 10e1092f-51d0-473a-80f8-137652fd0c39
+            - sddc_id: 15251c47-bad0-4adf-b8f2-85eba94a2a2f
+            - sddc_name: test-1234
+            - verify_ssl: False
+            - cert: /path/to/client/certificate
+
+.. warning::
+
+    It is recommended to pass the VMC authentication details using Pillars rather than specifying as plain text in SLS
+    files.
+
+"""
+import logging
+
+from saltext.vmware.utils import vmc_state
+
+log = logging.getLogger(__name__)
+
+
+def present(
+    name,
+    hostname,
+    refresh_key,
+    authorization_host,
+    org_id,
+    num_hosts,
+    provider,
+    region,
+    account_link_config=None,
+    account_link_sddc_config=None,
+    deployment_type=None,
+    host_instance_type=None,
+    msft_license_config=None,
+    sddc_id=None,
+    sddc_template_id=None,
+    sddc_type=None,
+    size=None,
+    skip_creating_vxlan=False,
+    sso_domain=None,
+    storage_capacity=None,
+    vpc_cidr=None,
+    vxlan_subnet=None,
+    validate_only=False,
+    verify_ssl=True,
+    cert=None,
+):
+    """
+    Ensure a given SDDC exists for given organization.
+
+    name
+        The name of SDDC that will be assigned to new created SDDC.
+
+    hostname
+        The host name of VMC.
+
+    refresh_key
+        API Token of the user which is used to get the Access Token required for VMC operations
+
+    authorization_host
+        Hostname of the VMC cloud console.
+
+    org_id
+        The ID of organization to which the SDDC belongs to.
+
+    num_hosts
+        An integer value to indicate number of hosts in a SDDC.
+
+    provider: String
+        Determines what additional properties are available based on cloud provider.
+        Possible values are: AWS, ZEROCLOUD
+
+    region: String
+        Aws region on which SDDC will be deployed.
+
+    account_link_config
+        (Optional) The account linking configuration, we will keep this one and remove accountLinkSddcConfig finally.
+
+        account_link_config expects to be passed as a dict having delay_account_link parameter
+            delay_account_link: Boolean
+            (Optional) Boolean flag identifying whether account linking should be delayed or not for the SDDC.
+
+            .. code::
+
+                {
+                    "delay_account_link": false
```

Comment:
**suggestion** YAML turns into Python dictionaries, so rather than give examples via dictionaries,:

```
.. code::

     account_link_config:
        delay_account_link: False
        other: parameters
        would: go here
```

Reply:
Updated

Reply:
Punctuation matters in YAML:

```
> Python 3.10.8 (main, Oct 13 2022, 21:13:48) [GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pprint, yaml
>>> pprint.pprint(yaml.safe_load('''
... account_link_config:
...   delay_account_link: False
... '''))
{'account_link_config': {'delay_account_link': False}}
>>> pprint.pprint(yaml.safe_load('''
... account_link_config:
...   - delay_account_link: False
... '''))
{'account_link_config': [{'delay_account_link': False}]}
>>> yaml.safe_load('''
... account_link_config
...   - delay_account_link: False
... ''')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.10/site-packages/yaml/__init__.py", line 125, in safe_load
    return load(stream, SafeLoader)
  File "/usr/lib/python3.10/site-packages/yaml/__init__.py", line 81, in load
    return loader.get_single_data()
  File "/usr/lib/python3.10/site-packages/yaml/constructor.py", line 49, in get_single_data
    node = self.get_single_node()
  File "/usr/lib/python3.10/site-packages/yaml/composer.py", line 36, in get_single_node
    document = self.compose_document()
  File "/usr/lib/python3.10/site-packages/yaml/composer.py", line 58, in compose_document
    self.get_event()
  File "/usr/lib/python3.10/site-packages/yaml/parser.py", line 118, in get_event
    self.current_event = self.state()
  File "/usr/lib/python3.10/site-packages/yaml/parser.py", line 193, in parse_document_end
    token = self.peek_token()
  File "/usr/lib/python3.10/site-packages/yaml/scanner.py", line 129, in peek_token
    self.fetch_more_tokens()
  File "/usr/lib/python3.10/site-packages/yaml/scanner.py", line 223, in fetch_more_tokens
    return self.fetch_value()
  File "/usr/lib/python3.10/site-packages/yaml/scanner.py", line 577, in fetch_value
    raise ScannerError(None, None,
yaml.scanner.ScannerError: mapping values are not allowed here
  in "<unicode string>", line 3, column 23:
      - delay_account_link: False
```

Also when it's YAML the code block should be 

```
.. code:: yaml

    account_link_config:
      delay_account_link: False
```

indents for YAML are 2 spaces by convention.

Reply:
Updated and verified